plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -40,11 +41,12 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 
 // implementation of the error interface
 func (f *FitError) Error() string {
-	output := fmt.Sprintf("failed to find fit for pod, ")
+	predicates := util.NewStringSet()
 	for node, predicateList := range f.FailedPredicates {
-		output = output + fmt.Sprintf("Node %s: %s", node, strings.Join(predicateList.List(), ","))
+		predicates = predicates.Union(predicateList)
+		glog.Infof("failed to find fit for pod %v on node %s: %s", f.Pod.Name, node, strings.Join(predicateList.List(), ","))
```

Comment:
seems fine here until we have a scheduler status page (and I think what you mean is more of a customized per-pod "why pending" page rather than a generic scheduler status page)